pkg/controller/machinemanagement/machinemanagement_controller.go

```diff
@@ -254,3 +271,26 @@ func (r *ReconcileMachineManagement) addAnnotationToTargetNamespace(cd *hivev1.C
 
 	return nil
 }
+
+func selectorSecretWatchHandler(a handler.MapObject) []reconcile.Request {
+	retval := []reconcile.Request{}
+
+	secret := a.Object.(*corev1.Secret)
+	if secret == nil {
+		// Wasn't a Secret, bail out. This should not happen.
+		log.Errorf("Error converting MapObject.Object to Secret. Value: %+v", a.Object)
+		return retval
+	}
+	if secret.Labels == nil {
+		return retval
+	}
+	cdName, ok := secret.Labels[constants.ClusterDeploymentNameLabel]
```

Comment:
So it looks like we're watching all Secrets just quickly discarding any without our label.

How can we effeciently do this now? Top of my head, maybe we annotate it with a json list of the target namespaces it should be synced to? This will not be fun to maintain though if someone is doing multiple CDs in one namespace re-using that Secret. Need to think on this a bit. Also horrible things people could do to fire Secrets into namespaces they have no business in...